hawc/apps/lit/api.py

```diff
@@ -166,6 +169,64 @@ def tag_heatmap(self, request, pk):
         export = FlatExport(df=df, filename=f"df-{instance.id}")
         return Response(export)
 
+    @transaction.atomic()
+    @action(
+        detail=True, methods=("patch",), url_path="replace-hero",
+    )
+    def replace_hero(self, request, pk):
+        assessment = self.get_object()
+        body = json.loads(request.body)
+        replace = body.get("replace", list())
+        replace_unzipped = list(zip(*replace))
+        ref_ids = replace_unzipped[0]
+        hero_ids = replace_unzipped[1]
+
+        # make sure all references are HERO and in assessment
+        matching_references = assessment.references.filter(
+            id__in=ref_ids, identifiers__database=constants.HERO
+        )
+        if matching_references.count() != len(ref_ids):
+            raise exceptions.ValidationError("All references must be from selected assessment.")
+
+        # set hero ref
+        for index, (ref, hero) in enumerate(replace):
```

Comment:
move all the code here into lit/models or lit/managers, and lit tasks.  the API should call biz logic, not have lots of it nested in it